Tests/FoundationEssentialsTests/FileManager/FileManagerTests.swift

```diff
@@ -331,6 +335,7 @@ final class FileManagerTests : XCTestCase {
     }
     
     func testLinkFileAtPathToPath() throws {
+        #if !os(Android) // Hard links are not normally allowed on Android.
```

Comment:
I don't think I see anything in the implementation that's specific to android with this function (unless that was in a previous PR). Are we missing an Android check in the implementation to throw? I think rather than avoiding the test if that's the case maybe it makes sense to `XCTAssertThrowsError` here?

Reply:
If calling `link()` doesn't work on Android, I think we should replace that call in the implementation to something that will work on Android, or if Android does not have that concept then we should probably throw to indicate that it did not succeed. That way, either the test will pass as-is or we can update the test to check for the thrown error on Android

Reply:
[Android doesn't allow hard linking on most partitions for regular users for the last decade](https://stackoverflow.com/questions/32365690/is-android-m-not-allowing-hard-links). Since it may work in those few remaining scenarios, I didn't really want to think about it and just disabled the test.

Since it will usually fail, I could replace it with this if you'd like instead:
```
#if os(Android)
XCTAssertEqual($0.delegateCaptures.shouldProceedAfterLinkError, [.init("foo", "bar", code: .fileWriteNoPermission)])
XCTAssertFalse($0.fileExists(atPath: "bar"))
#else
... current passing asserts ...
```

Reply:
As discussed, added those Android-specific asserts to this pull